sw/device/lib/runtime/hart.c

```diff
@@ -10,9 +10,11 @@
 
 extern void wait_for_interrupt(void);
 
-void busy_sleep_micros(size_t microseconds) {
-  size_t cycles = kIbexClockFreqHz * microseconds / 10000000;
-  ibex_busy_loop(cycles);
+void usleep(uint32_t usec) {
+  uint64_t cycles = (uint64_t)kIbexClockFreqHz * usec / 1000000;
+  uint64_t start = read_cycle_counter();
+  while ((uint64_t)(read_cycle_counter() - start) < cycles) {
```

Comment:
Isn't that cast unnecessary?

Reply:
Yes, (in this case, I think, not if we used something smaller like `uint16_t`) but I can't always remember all the integer promotion rules so I use the cast to be safe. I can definitely remove it if you want.

Reply:
Removed.